Allow lambda@edge in aws-lambda-function module

palasha opened the pull request:
I want to use this module to create lambdas for cloudfront. That
requires supporting lambda@Edge. The two main changes here are:
1. Adding edgelambda as an authorized service for the role
2. granting permission create/update permission on * log groups.

The second is I suppose slightly controversial. because its an edge
lambda, we would have to precreate the log groups in every single aws
region, and I have no clue how to do that. I think this is an okay
tradeoff for now, and tried to document that in the description.

Test Plan:
wait for CI
use it in stacked TF and see if it works.

palasha commented:
make fmt to fix lint

palasha commented:
boolean -> bool

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/44dfb03f1a68c37245ae7a5ce9d81e5bbdfc0819..d4d6e0ebf1a3667d7b7af74239b2369d941ea4b6).

cziie commented on aws-cloudfront-naked-domain-redirect/variables.tf:
```diff
@@ -0,0 +1,19 @@
+  description = "Env for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable service {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "service" is declared but not used

cziie commented on aws-cloudfront-naked-domain-redirect/variables.tf:
```diff
@@ -0,0 +1,19 @@
+  description = "Service for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable owner {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "owner" is declared but not used

cziie commented on aws-cloudfront-naked-domain-redirect/variables.tf:
```diff
@@ -0,0 +1,19 @@
+variable project {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "project" is declared but not used

cziie commented on aws-cloudfront-naked-domain-redirect/variables.tf:
```diff
@@ -0,0 +1,19 @@
+  description = "Project for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable env {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "env" is declared but not used

cziie commented on aws-lambda-edge-add-hsts-header/main.tf:
```diff
@@ -0,0 +1,16 @@
+data "archive_file" "lambda" {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Missing version constraint for provider "archive" in "required_providers"

cziie replied:
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
data "archive_file" "lambda" is declared but not used

cziie commented on aws-lambda-edge-add-hsts-header/variables.tf:
```diff
@@ -0,0 +1,19 @@
+  description = "Env for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable service {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "service" is declared but not used

cziie commented on aws-lambda-edge-add-hsts-header/variables.tf:
```diff
@@ -0,0 +1,19 @@
+  description = "Service for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable owner {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "owner" is declared but not used

cziie commented on aws-lambda-edge-add-hsts-header/variables.tf:
```diff
@@ -0,0 +1,19 @@
+variable project {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "project" is declared but not used

cziie commented on aws-lambda-edge-add-hsts-header/variables.tf:
```diff
@@ -0,0 +1,19 @@
+  description = "Project for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable env {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
variable "env" is declared but not used

cziie commented on aws-lambda-function/outputs.tf:
```diff
@@ -2,6 +2,10 @@ output arn {
   value = aws_lambda_function.lambda.arn
 }
 
+output qualified_arn {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`qualified_arn` output has no description

cziie commented on aws-lambda-function/outputs.tf:
```diff
@@ -2,6 +2,10 @@ output arn {
+  value = aws_lambda_function.lambda.qualified_arn
+}
+
 output invoke_arn {
```
**[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`invoke_arn` output has no description

palasha commented:
add publish argument and fix lint

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/a199d9534425024fd05f374dc63775437531df51..ff72a9cbe71d29c24d9ca1123d5fe94774c65694).

palasha commented:
update README

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/7f37decedab836deafb8f746121e06c302a8c121..6d93c590224136065fe39d399c6ccfd6ea9a0713).

ryanking commented:
@palasha given that your test plan includes some manual testing, can you let me know when that is done?

palasha commented:
> @palasha given that your test plan includes some manual testing, can you let me know when that is done?

Sorry for the delay here! The manual testing is that:
- this actively working in production for the comealong.org website to add security headers.
- I've verified that no cloudwatch logs are being created in different regions.

palasha commented:
> @palasha given that your test plan includes some manual testing, can you let me know when that is done?

Sorry for the delay here! The manual testing is that:
- this actively working in production for the comealong.org website to add security headers.
- I've verified that no cloudwatch logs are being created in different regions.
